src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/ProjectDebuggerProvider.cs

```diff
@@ -118,7 +126,16 @@ public static bool IsDotNetCoreFramework(string targetFramework)
             var launchProvider = GetLaunchTargetsProvider(activeProfile) ??
                 throw new Exception(string.Format(VSResources.DontKnowHowToRunProfile, activeProfile.Name));
 
-            var launchSettings = await launchProvider.QueryDebugTargetsAsync(launchOptions, activeProfile).ConfigureAwait(true);
+            IReadOnlyList<IDebugLaunchSettings> launchSettings;
+            if (fromDebugLaunch && launchProvider is IDebugProfileLaunchTargetsProvider2 launchProvider2)
```

Comment:
I'm a bit confused by the delegation order here in the sense that the default implementation will still launch processes.  Wouldn't it be better for the delegation to go the other way? (So that the new method is the one that does the new thing? Returning targets but not starting processes).

Reply:
Can't because of compatibility. For most cases, including the console application case in this project system, calling the old method is fine as it doesn't start anything. It is only if you need to distinguish what happens between the two cases (like web does) that  you need the new interface.. This is why I included forDebugLaunch in the name of the new method.